erpnext/stock/get_item_details.py

```diff
@@ -849,6 +852,7 @@ def insert_item_price(args):
 						"item_code": args.item_code,
 						"currency": args.currency,
 						"price_list_rate": price_list_rate,
+						"uom": args.stock_uom,
```

Comment:
I will add a test but If I understand you correctly you are saying that if the user adds 1 box that has 100 Nos.
Price will be added for Nos instead of Box but this is the intended behavior as you can see [here](https://github.com/frappe/erpnext/blob/3abd947982e45816109d0024a208436317a0ab94/erpnext/stock/get_item_details.py#L826) price is also converted as per the stock_uom by `(args.rate + args.discount_amount) / args.get("conversion_factor")` and it will be converted to sales UOM at the time of next SI Creation [ box or any other] as intended. 

Example:
Price of 1 Nos = 1 Box [1000₹] / Nos in Box [100] 
1 Nos = 10 ₹

Price_List_Rate is added as below
(
  item: any_product,
  UOM: "Nos",
  Price: 10 ₹
)
If I misunderstood your point can you please elaborate?

Reply:
@maharshivpatel looks good. I didn't see the change in the context of how the price list rate is being derived right now. 


Perheps this can be changed later someday to support inserting rates based on UOM only.